pipeline.go

```diff
@@ -51,11 +56,75 @@ func setup(path string) error {
 }
 
 //Run executes the pipeline
-func Run(pipeline Pipeline) ([]StageExecutionResult, error) {
+func (p *Pipeline) Run(pipeline Pipeline) ([]StageExecutionResult, error) {
 	if err := setup(pipeline.DefaultRepo); err != nil {
 		return nil, fmt.Errorf("error in inicial setup. %q", err)
+	}
+
+	for _, stage := range pipeline.Stages {
+		var er StageExecutionResult
+		var err error
 
+		id := fmt.Sprintf("%s-%s-%s", stage.Name, pipeline.DefaultEnv["month"], pipeline.DefaultEnv["year"])
+		log.Printf("Executing %s ...\n", id)
+
+		er, err = build(stage)
+		if err != nil {
+			return nil, fmt.Errorf("error in image build %s", err)
+		}
+
+		fmt.Println(er)
 	}
 
 	return []StageExecutionResult{}, nil
 }
+
+// Build tries to build a docker image for a job and panics if it can not suceed.
+func build(s Stage) (StageExecutionResult, error) {
+	id := fmt.Sprintf("%s", s.Name)
+	log.Printf("Building image %s...\n", id)
```

Comment:
Bulding Stage?

Reply:
Adicionei: "Building image for stage %s"